core/domain/suggestion_services.py

```diff
@@ -1826,3 +1828,805 @@ def update_question_suggestion(
+        suggestion_models.TranslationReviewStatsModel.get(
+            suggestion.change.language_code, user_id, topic_id
+        ))
+
+    if translation_review_stat_model is None:
+        suggestion_models.TranslationReviewStatsModel.create(
+            language_code=suggestion.change.language_code,
+            reviewer_user_id=user_id,
+            topic_id=topic_id,
+            reviewed_translations_count=1,
+            reviewed_translation_word_count=content_word_count,
+            accepted_translations_count=1,
+            accepted_translations_with_reviewer_edits_count=1 * int(
+                suggestion.edited_by_reviewer),
+            accepted_translation_word_count=content_word_count * int(
+                is_accepted),
+            first_contribution_date=(
+                datetime.datetime.strptime(
+                    str(
+                        suggestion.last_updated.utcnow().date().isoformat()),
+                        '%Y-%m-%d'
+                    ).date()),
+            last_contribution_date=(
+                datetime.datetime.strptime(
+                    str(
+                        suggestion.last_updated.utcnow().date().isoformat()),
+                        '%Y-%m-%d'
+                    ).date())
+        )
+    else:
+        translation_review_stat = (
+            _create_translation_review_stats_from_model(
+                translation_review_stat_model))
+
+        translation_review_stat.reviewed_translations_count += 1
+        translation_review_stat.reviewed_translation_word_count += (
+            content_word_count)
+        translation_review_stat.accepted_translations_count += 1 * int(
+            is_accepted)
+        (
+            translation_review_stat
+            .accepted_translations_with_reviewer_edits_count
+        ) += 1 + int(suggestion.edited_by_reviewer)
+        translation_review_stat.last_contribution_date = (
+                datetime.datetime.strptime(
+                    str(
+                        suggestion.last_updated.utcnow().date().isoformat()),
+                        '%Y-%m-%d'
+                    ).date())
+
+        _update_translation_review_stats_models([translation_review_stat])
+
+    update_translation_contribution_stats_at_review(
+        suggestion, suggestion.author_id, target_id)
+
+
+def update_question_contribution_stats_at_submission(
+    suggestion: suggestion_registry.BaseSuggestion,
+    user_id: str, target_id: str) -> None:
+    """Creates/updates QuestionContributionStatsModel model for given question
+    submitter when a question is submitted.
+
+    Args:
+        suggestion: Suggestion. The suggestion domain object that is being
+            submitted.
+        user_id: str. The ID of the corresponding submitter.
+        target_id: str. The ID of the target entity being suggested to.
+    """
+    topic_ids = []
+    topics = skill_services.get_all_topic_assignments_for_skill(target_id) # type: ignore[no-untyped-call]
```

Comment:
@seanlip Sagang confirmed that, a skill will be assigned to only one topic. But in any case if there would be more than one topics that have a given skill, there could be fake contributor/reviewer statistics created. In that case shall we store question contribution/review stats per skill per user.(Currently we are storing question contribution/review stats per topic per user).

If we store question contribution/review stats per skill per user, then we will have to update this mock to display stats per skill.(Currently we have planned to show per topic https://www.figma.com/file/SNOHAatFcCvdvIAdxAzF6E/Contributor-Dashboard-Achievements?node-id=112%3A40)

cc/ @sagangwee

Reply:
Wait ... I'm pretty sure that, in our current system for content management, a skill can appear in multiple topics. So it's important to account for that.

I would say that, in the mocks, we should still show "per topic" since that's the milestone that matters. Also, if you show skills, it will overflow very quickly. I have no problems with a contribution being counted for two topics if the skill is in two topics, that is totally OK.

Whether to store stats for skills in the backend (in case it's needed later) is a question I'm not sure about. Do you have any thoughts on this? One possible issue is that there is the potential for "score decreases" if we change the assignment of skills to topics later. On the other hand I can imagine having 1000 skills and I'm not sure if that will result in latency issues when fetching the data. Additionally it's possible to merge skills into other skills and we'd need to deal with that properly in the future.

On the whole, I would say, if we don't ever plan to show stats for skills, we probably don't need to do the bookkeeping for that.

Reply:
If a contribution being counted for two topics if the skill is in two topics, is fine I don't think that there is an issue for going ahead with the current plan as well. We can keep the storage models as it is right now(Also we have not planned to show stats for skills before as well).

"score decreases" should not happen even if skill assignments are changed because at the submission/review of the suggestion we store the stats data in the stats model with corresponding topic id that the skill was assigned at the time of submission/review. Those data will not be changed even if the skill assignments are changed.

So I think, we can go ahead with currently what we have done in the storage layer by storing question stats per topic per user.